07_array_algos/practice/solutions.cpp

```diff
@@ -0,0 +1,357 @@
+/****************************************************************************
+ * This file is part of the "Introduction to programming" course. FMI 2019/20
+ *****************************************************************************/
+
+ /**
+  * @file   solutions.cpp
+  * @author Ivan Filipov
+  * @author Kristian Krastev
+  * @date   12.2019
+  * @brief  Solution for function tasks from practice 7.
+  */
+
+#include <iostream>
+#include <random>
+#include <time.h>
+
+const size_t MAX_N = 1024; // max number of elements we can store
+const size_t SORTED_SIZE = 6;
+const size_t MAX_ROWS = 5;
+const size_t MAX_COLS = 5;
+const size_t BOYS_SIZE = 10;
+const size_t GIRLS_SIZE = 10;
+
+// task_01
+bool descending_array(int arr[], size_t size) {
+
+    for (size_t i = 0; i < size - 1; i++) {
+        
+        if (arr[i] < arr[i + 1])
+            return false;
+    }
+    
+    return true;
+}
+
+// task_02
+// int because we want return the new size
+int insert_elem(int arr[], size_t size, int elem) {
+
+    if (size >= MAX_N) 
+        return size;
+
+    // if empty
+    if (size == 0) {
+        arr[0] = elem;
+        return 1;
+    }
+  
+    // we traverse backwards in order to move the elements one step to the right
+    // to make room for our new element
+    int i;
+    for (i = size - 1; (i >= 0 && arr[i] > elem); i--) {
+        arr[i + 1] = arr[i]; 
+    }
+  
+    arr[i + 1] = elem;
+    return size + 1;
+}
+
+// task_03
+// because we want to return the updated size
+int remove_elem(int arr[], size_t size, int elem) {
+
+    if (size < 1) {
+        std::cout << "There are no elements here :( \n";
+        return size;
+    }
+    size_t i = 0;
+    
+    while ((i < size) && (arr[i++] != elem)) {} // to get the position of the element
+    
+    if (i >= size) {
+        std::cout << "This element does not exist in the given array!\n";
+        return size;
+    }
+
+    // to override the value of our element
+    while (i < size) {
+        arr[i - 1] = arr[i];
+        ++i;
+    }
+
+    return size - 1;
+}
+
+// task_04
+bool is_symetric(int arr[], size_t size) {
+
+    size_t middle = size / 2;
+    size_t i = 0;
+
+    while ((i < middle) && (arr[i] == arr[size - 1 - i])) {
```

Comment:
`for (size_t i = 0; i < middle; i++)
     if (arr[i] != arr[size - 1 - i])
        return false;
`
....

`return true;`